### ISSUE TITLE
Request - change operator username

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#8813]
Help us understand your request (check below):

- [x] other

**Describe what you're trying to do** 

There have been a couple requests to change an Operator's username recently.

Creating a new/second account would, I've been told, violate TACC's policies, although perhaps they'd be OK with some sort of permanent disabling and redirect (eg something that's clearly **not** one person with two access points). This seems like the simplest option in some ways, but it would also greatly complicate the internal link between operators and agents, and that would raise significant security concerns. (But I sometimes wonder if our entire operator invitation procedure shouldn't be revisited anyway...)

The username is used as a key in lots of places (current list below), so simply changing username would have some potential to be cryptic and/or lossy.

A new user adopting the old username (which would not be prevented by a simple change) would provide a path for 'someuser' in some archive to mean two things, which is a bit beyond 'cryptic.'

Some of these references are to tools, but even then things like the bulkloader are kept and can be referred to long after entry is complete.
Others are internal logs, although converting these to keys could work. (And I'd guess there are logs which aren't properly keyed, the list looks short...)

Possibly some sort of ``cf_users.previous_username`` and the code to consider it could work, but there's no scalability in that.

I don't see a good or easy way to approach this, but I suspect we have to find a way to do it anyway. 

Possibly I'm just being too paranoid and that all of these concerns are theoretical and can be dismissed....

Help??

```
arctosprod@arctos>> \d cf_users
                                                    Table "core.cf_users"
           Column            |            Type             | Collation | Nullable |                 Default                  
-----------------------------+-----------------------------+-----------+----------+------------------------------------------
 username                    | character varying(30)       |           | not null | 
 password                    | character varying(255)      |           | not null | 
 displayrows                 | integer                     |           |          | 
 user_id                     | integer                     |           | not null | nextval('sq_cf_users_user_id'::regclass)
 customotheridentifier       | character varying(255)      |           |          | 
 resultcolumnlist            | character varying(4000)     |           |          | 
 pw_change_date              | timestamp without time zone |           | not null | 
 last_login                  | timestamp without time zone |           |          | 
 customoidoper               | character varying(255)      |           |          | 
 taxapickprefs               | character varying(255)      |           |          | 
 sdmapclass                  | character varying(30)       |           |          | 
 auth_key                    | character varying           |           |          | 
 auth_key_expires            | timestamp without time zone |           |          | 
 failcount                   | integer                     |           |          | 
 partview                    | character varying(40)       |           |          | 
 browse_bulk_rows            | smallint                    |           |          | 10
 idsview                     | character varying(40)       |           |          | 
 taxapicksource              | character varying           |           |          | 
 first_name                  | character varying(60)       |           |          | 
 middle_name                 | character varying(60)       |           |          | 
 last_name                   | character varying(60)       |           |          | 
 affiliation                 | character varying(255)      |           |          | 
 email                       | character varying(255)      |           |          | 
 ask_for_filename            | smallint                    |           |          | 1
 request_communication       | boolean                     |           |          | false
 place_search_rows           | smallint                    |           |          | 10
 geog_rslt_cols              | character varying           |           |          | 
 loc_rslt_cols               | character varying           |           |          | 
 evnt_rslt_cols              | character varying           |           |          | 
 geog_srch_cols              | character varying           |           |          | 
 loc_srch_cols               | character varying           |           |          | 
 evnt_srch_cols              | character varying           |           |          | 
 catrec_rslt_cols            | character varying           |           |          | 
 catrec_srch_cols            | character varying           |           |          | 
 directory_view              | character varying           |           |          | 
 loanpickcols                | character varying           |           |          | 
 include_verbatim            | character varying(10)       |           |          | 
 data_entry_related_settings | jsonb                       |           |          | 
 data_entry_profile_key      | integer                     |           |          | 
 loan_item_review_rows       | integer                     |           |          | 
 operator_agent_id           | integer                     |           |          | 
 reporter_prefs              | character varying           |           |          | 
 last_recovery               | timestamp without time zone |           |          | 
Indexes:
    "cf_users_pkey" PRIMARY KEY, btree (user_id)
    "iu_cf_users_operator_agent_id" UNIQUE, btree (operator_agent_id)
    "iu_u_username" UNIQUE, btree (upper(username::text))
    "ix_cf_users_username" UNIQUE, btree (username)
    "ix_l_username" btree (lower(username::text))
Check constraints:
    "check_affiliation" CHECK (check_user_input(affiliation))
    "check_first_name" CHECK (check_user_input(first_name))
    "check_last_name" CHECK (check_user_input(last_name))
    "check_middle_name" CHECK (check_user_input(middle_name))
    "check_username" CHECK (check_username(username))
    "ck_valid_email" CHECK (ck_isemail(email::text))
Foreign-key constraints:
    "cf_users_operator_agent_id_fkey" FOREIGN KEY (operator_agent_id) REFERENCES agent(agent_id)
Referenced by:
    TABLE "bulkloader" CONSTRAINT "bulkloader_enteredby_fkey" FOREIGN KEY (enteredby) REFERENCES cf_users(username)
    TABLE "cf_temp_attributes" CONSTRAINT "cf_temp_attributes_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_barcodeswapper" CONSTRAINT "cf_temp_barcodeswapper_fk_user" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_citation" CONSTRAINT "cf_temp_citation_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_container_environment" CONSTRAINT "cf_temp_container_environment_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_container_to_position" CONSTRAINT "cf_temp_container_to_position_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_getcontainer" CONSTRAINT "cf_temp_getcontainer_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_move_container" CONSTRAINT "cf_temp_move_container_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_part_attribute_unloader" CONSTRAINT "cf_temp_part_attribute_unloader_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_parts" CONSTRAINT "cf_temp_parts_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_taxon_check" CONSTRAINT "cf_temp_taxon_check_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_unload_citation" CONSTRAINT "cf_temp_unload_citation_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_user_invite" CONSTRAINT "cf_temp_user_invite_invited_by_username_fkey" FOREIGN KEY (invited_by_username) REFERENCES cf_users(username)
    TABLE "bulkloader_stage" CONSTRAINT "fk_bulkloader_stage_enteredby" FOREIGN KEY (enteredby) REFERENCES cf_users(username)
    TABLE "cf_cat_rec_srch_profile" CONSTRAINT "fk_cf_cat_rec_srch_profile_cfusers" FOREIGN KEY (cf_username) REFERENCES cf_users(username)
    TABLE "cf_temp_taxon_name" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_specevent" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_dataloan_add" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_dataloan_remove" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_edit_container" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_container_create" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_cat_rec_rmk" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_accn" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_collector" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_media" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_taxon_relation" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_unload_identifiers" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_identifier_converter" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_media_labels_ldr" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_obj_encumbrance" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_barcodeswapper" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_event_attrs" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_oids" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_spec_part_attr" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_collecting_event" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_locality" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_edit_locality" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_barcode_parts" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_loan_item" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_media_relations_ldr" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_loan" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_geog_lookup" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_bulk_edit_parts" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_bulk_identifier_issuedby" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "cf_temp_locality_attributes" CONSTRAINT "fk_username" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctcollection_attribute_type" CONSTRAINT "log_ctcollection_attribute_type_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctlanguage" CONSTRAINT "log_ctlanguage_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctlicense" CONSTRAINT "log_ctlicense_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctlocality_attribute_type" CONSTRAINT "log_ctlocality_attribute_type_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctmortality_cause" CONSTRAINT "log_ctmortality_cause_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctpart_attribute_type" CONSTRAINT "log_ctpart_attribute_type_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctterms" CONSTRAINT "log_ctterms_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctuse_condition" CONSTRAINT "log_ctuse_condition_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)
    TABLE "log_ctwater_body" CONSTRAINT "log_ctwater_body_username_fkey" FOREIGN KEY (username) REFERENCES cf_users(username)

```

**Comment by dustymc** [comment]
@destinyfgonzalez can you attend some AWG meeting to help us prioritize/resolve this?

**Comment by mkoo** [comment]
emailed Destiny directly as this seems to be a pretty specific user issue
